### PR TITLE
feat: update deps and use vitest

### PR DESCRIPTION
**Pull request by spaenleh**
- use vitest
- update eslint to flat config

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        'error',
+        {
+          allowExpressions: true,
+          allowHigherOrderFunctions: true,
+          allowTypedFunctionExpressions: true,
+        },
+      ],
+
+      //   '@typescript-eslint/no-var-requires': 'off',
+      //   'global-require': 'off',
```
Comment (pyphilia) [inline eslint.config.mjs:123]: Remove?

File: src/api/appData.ts [hunk 2]
```diff
@@ -95,7 +95,10 @@ export const getAppDataFile = async ({
   return axios
     .get<Blob>(url, {
       responseType: 'blob',
-      withCredentials: false,
+      headers: {
+        'Access-Control-Allow-Credentials': true,
+      },
+      // withCredentials: false,
```
Comment (pyphilia) [inline src/api/appData.ts:101]: Remove

**Review by pyphilia — APPROVED** [review]
LGTM 🎻

**Review by ReidyT — APPROVED** [review]
Nice PR, LGTM!

**Pull request merged**